File: Ejercicios JAVASCRIPT/Clase 9/Persona.js
class Persona {

        static contadorPersona = 0;

        constructor(nombre, apellido, edad) {
                this._idPersona = ++Persona.contadorPersona;
                this._nombre = nombre;
                this._apellido = apellido;
                this._edad = edad;
        }
        get idPersona() {
                return this._idPersona;
        }
        get nombre() {
                return this._nombre;
        }
        get apellido() {
                return this._apellido;
        }
        get edad() {
                return this._edad;
        }
        set nombre(nombre) {
                this._nombre = nombre;
        }
        set apellido(apellido) {
                this._apellido = apellido;
        }
        set edad(edad) {
                this._edad = edad;
        }
        toString() {
                return `${this._idPersona} ${this._nombre} ${this._apellido} ${this._edad}`;
        }
}

File: Ejercicios JAVA/Clase 2/BloquesInicializacion/src/domain/Persona.java
package domain;

public class Persona {
    
    private final int idPersona;           
    private static int contadorPersonas;    

    static{
        System.out.println("Ejecucion del Bloque de inicializacion estatico");
        ++Persona.contadorPersonas; 
    }
       {
        System.out.println("Ejecucion del bloque no estatico");
        this.idPersona = ++Persona.contadorPersonas; 
    }
    public Persona(){
        System.out.println("Ejecucion del constructor");
    }
    public int getIdPersona(){       
        return this.idPersona;       
    }
    @Override
    public String toString() {
        return "Persona{" + "idPersona=" + idPersona + '}';
    }
}
